docs/design/audit-v2.md

```diff
@@ -0,0 +1,93 @@
+# Auditing V2: Random Node Selection
+
+## Abstract
+
+This design doc outlines how we will implement a Version 2 of the auditing service.
+With our current auditing service, we are auditing per segment of data.
+We propose to replace this method of auditing with a new selection process that selects by random node instead.
+
+## Background
+
+As our network grows, it will take longer for nodes to get vetted.
+This is because every time an upload happens, we send 5% of the uploaded data to unvetted nodes and 95% to vetted nodes.
+When auditing occurs, we currently select a random stripe within a random segment.
+If we're selecting audits at random per byte, every segment has some percentage that went to vetted nodes.
+As more nodes join the network, it will become exponentially less likely that an unvetted node will be audited since most data will be stored on vetted nodes.
+With a satellite with one petabyte of data, new nodes will take one month to get vetted. However, with 12PB of data on the network, this vetting process time would take 12 months, which is much too long. We need a scalable approach.
+
+We want a way to select segments to audit based on statistically randomly selected storage nodes.
+Currently, there's not a way to select a random storage node and audit on that basis.
+
+## Design
+
+We will create an audit observer that uses the metainfo loop, and this observer will create a reservoir sample of segments for every node. This audit observer will replace the existing method of auditing per byte. The audit will pick a random node, then a random segment from the node's reservoir.
```

Comment:
"the audit will pick a random node" should probably be "a configurable number of random nodes"

Reply:
okay- so "The audit will pick a configurable number of random nodes, then a random segment from one of the nodes' reservoirs"?

Reply:
I'm just saying that the wording is vague.  Do we loop through all of metainfo db and build a reservoir cache for each node only to audit a single one of them?  Probably not.  We probably want to audit (a random segment each) from _MANY_ of them.

Reply:
ahh that totally makes sense, will clarify.